Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp

```diff
@@ -252,15 +263,22 @@ void ProcessLauncher::launchProcess()
     if (!process.get())
         g_error("Unable to spawn a new child process: %s", error->message);
 
+#if OS(LINUX)
+    m_processID = IPC::readPIDFromPeer(pidSocketPair.server);
+
+    RELEASE_ASSERT(!close(pidSocketPair.server));
+    RELEASE_ASSERT(!close(pidSocketPair.client));
```

Comment:
Wasn't this ownership transferred to GSubprocessLauncher?

Reply:
It doesn't fail because it's closed for the first time here. Later on, GSubprocessLauncher will try to close the same integer again, ignoring errors. This could close an innocent unrealted fd.